parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -2,144 +2,154 @@ package hmda.parser.fi.lar
 
 import hmda.model.fi.lar._
 
-import scala.util.Try
+import scala.collection.immutable.ListMap
+import scalaz._
+import scalaz.Scalaz._
+import scala.util.{ Failure, Success, Try }
 
 object LarCsvParser {
   def apply(s: String): Either[List[String], LoanApplicationRegister] = {
-    val larErrors = checkLar(s)
+    val values = s.split('|').map(_.trim)
+    val parserResults = checkLar(values.toList)
+    parserResults match {
+      case scalaz.Success(convertedValues) => {
 
-    if (larErrors.isEmpty) {
-      val values = s.split('|').map(_.trim)
-      val id = values(0).toInt
-      val respId = values(1)
-      val agencyCode = values(2).toInt
-      val loanId = values(3)
-      val loanDate = values(4)
-      val loanType = values(5).toInt
-      val propertyType = values(6).toInt
-      val loanPurpose = values(7).toInt
-      val occupancy = values(8).toInt
-      val loanAmount = values(9).toInt
-      val preapprovals = values(10).toInt
-      val actionType = values(11).toInt
-      val actionDate = values(12).toInt
-      val msa = values(13)
-      val state = values(14)
-      val county = values(15)
-      val tract = values(16)
-      val appEthnicity = values(17).toInt
-      val coAppEthnicity = values(18).toInt
-      val appRace1 = values(19).toInt
-      val appRace2 = values(20)
-      val appRace3 = values(21)
-      val appRace4 = values(22)
-      val appRace5 = values(23)
-      val coAppRace1 = values(24).toInt
-      val coAppRace2 = values(25)
-      val coAppRace3 = values(26)
-      val coAppRace4 = values(27)
-      val coAppRace5 = values(28)
-      val appSex = values(29).toInt
-      val coAppSex = values(30).toInt
-      val appIncome = values(31)
-      val purchaserType = values(32).toInt
-      val denial1 = values(33)
-      val denial2 = values(34)
-      val denial3 = values(35)
-      val rateSpread = values(36)
-      val hoepaStatus = values(37).toInt
-      val lienStatus = values(38).toInt
+        val id = convertedValues(0)
+        val respId = values(1)
+        val agencyCode = convertedValues(1)
+        val loanId = values(3)
+        val loanDate = values(4)
+        val loanType = convertedValues(2)
+        val propertyType = convertedValues(3)
+        val loanPurpose = convertedValues(4)
+        val occupancy = convertedValues(5)
+        val loanAmount = convertedValues(6)
+        val preapprovals = convertedValues(7)
+        val actionType = convertedValues(8)
+        val actionDate = convertedValues(9)
```

Comment:
I don't like that the order of these fields appears twice in the code: both here and in `checkLar`. I don't want to hold up the decision because of it, and I don't yet have a better idea, but I do want to mention it. (Perhaps it'll inspire someone else!)